core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -327,8 +326,14 @@ object SparkSubmit {
       printStream.println("\n")
     }
 
-    val loader = new ExecutorURLClassLoader(new Array[URL](0),
-      Thread.currentThread.getContextClassLoader)
+    val loader =
+      if (sysProps.getOrElse("spark.driver.userClassPathFirst", "false").toBoolean) {
+        new ChildFirstURLClassLoader(new Array[URL](0),
```

Comment:
i think we just use ChildExecutorURLClassLoader and ExecutorURLClassLoader that it has existed. so we reduce code to change.